static/app/components/button.tsx

```diff
@@ -30,6 +33,10 @@ interface BaseButtonProps
    * Positions the text within the button.
    */
   align?: 'center' | 'left' | 'right';
+  /**
+   * Adds extra parameters to the analytics tracking
+   */
+  analyticsParams?: {};
```

Comment:
Typing this as an empty object won't allow users to pass anything as typescript will complain about properties that don't exist. `Record<string, any>` would allow anything to be passed, but the _real_ correct typing is what we have in `makeAnalyticsFunction` which uses [generics](https://www.typescriptlang.org/docs/handbook/2/generics.html) to change what params are accepted based on what you pass for the `eventKey`. It's a bit challenging, LMK if you need help on this 😄